### PR TITLE
HPCC-9395 Expand hashjoin to hashdistributes+local join.

### PR DESCRIPTION
**Pull request by jakesmith**
Also adjust resourcing of DISTRIBUTE, allowing for greater than 2.
At the moment, with the other hard limits, the practical limit is
increased to 4.

Signed-off-by: Jake Smith jake.smith@lexisnexis.com

### CODE REVIEW

**Comment by jakesmith** [comment]
@ghalliday - please review

File: ecl/hqlcpp/hqlcpp.cpp [hunk 1]
```diff
@@ -1629,7 +1629,7 @@ void HqlCppTranslator::cacheOptions()
         DebugOption(options.resourceMaxSockets, "resourceMaxSockets", 0),
         DebugOption(options.resourceMaxActivities, "resourceMaxActivities", 0),
         DebugOption(options.resourceMaxHeavy, "resourceMaxHeavy", 1),
-        DebugOption(options.resourceMaxDistribute, "resourceMaxDistribute", 2),
+        DebugOption(options.resourceMaxDistribute, "resourceMaxDistribute", 8),
```
Comment (jakesmith) [inline ecl/hqlcpp/hqlcpp.cpp]: As I mention in the commit message, the practical limit is 4 for the time being, due to the fixed memory constraints imposed by the fixed cluster size (see: HPCC-12936)

**Comment by ghalliday** [comment]
Have you run any tests to compare the performance?

**Comment by jakesmith** [comment]
Yes, on my 10x, it was about 10% faster with 2 * distributes + local join vs hashjoin.

**Comment by ghalliday** [comment]
Any ideas why?

**Comment by jakesmith** [comment]
probably, because there's more asynchronicity.. join, pull 1 side with a lookahead
I suspect a large test may have equalled out to negate the difference.

**Pull request merged**